synapseclient/core/credentials/credential_provider.py

```diff
@@ -52,28 +59,20 @@ def _get_auth_info(
         return None, None
 
     def get_synapse_credentials(
-        self, syn: "Synapse", user_login_args: Dict[str, str]
+        self, syn: "Synapse", user_login_args: UserLoginArgs
     ) -> Union[SynapseCredentials, None]:
-        """
-        Returns
-        [SynapseCredentials][synapseclient.core.credentials.cred_data.SynapseCredentials]
-        if this provider is able to get valid credentials, returns None otherwise.
 
-        Arguments:
-            syn: Synapse client instance
-            user_login_args: subset of arguments passed during syn.login()
+        if not user_login_args.auth_token and not user_login_args.username:
+            auth_profiles = get_config_authentication(syn.configPath, user_login_args.profile)
+            user_login_args.username = auth_profiles["username"]
+            user_login_args.auth_token = auth_profiles["auth_token"]
```

Comment:
Te should probably only check for `if not user_login_args.auth_token`.

And then only set the `user_login_args.username`, `if not user_login_args.username`